### PR TITLE
chore: add godoc to packet server handlers

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

I also decided to remove comments which I felt provided no value

closes: #7148

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/packet-server/keeper/keeper.go [hunk 1]
```diff
@@ -124,7 +134,6 @@ func (k Keeper) RecvPacket(
 
 	// Lookup counterparty associated with our channel and ensure that it was packet was indeed
 	// sent by our counterparty.
-	// Note: This can be implemented by the current keeper
```
Comment (colin-axner) [inline modules/core/packet-server/keeper/keeper.go:127]: I'm not sure what this was supposed to indicate?

File: modules/core/packet-server/keeper/keeper.go [hunk 2]
```diff
@@ -101,16 +107,20 @@ func (k Keeper) SendPacket(
...
 
-	// return the sequence
 	return sequence, nil
 }
 
-// TODO: add godoc
+// RecvPacket implements the packet receiving logic required by a packet handler.
+// The packet is checked for correctness including asserting that the packet was
+// sent and received on clients which are counterparties for one another.
+// If the packet has already been received an no-op error is returned.
```
Comment (AdityaSripal) [inline modules/core/packet-server/keeper/keeper.go:120]:
```suggestion
// If the packet has already been received a no-op error is returned.
```

File: modules/core/packet-server/keeper/keeper.go [hunk 3]
```diff
@@ -174,16 +183,18 @@ func (k Keeper) RecvPacket(
...
 	channelkeeper.EmitRecvPacketEvent(ctx, packet, sentinelChannel(packet.DestinationChannel))
 
 	return packet.AppVersion, nil
 }
 
-// TODO: add godoc
+// WriteAcknowledgement implements the async acknowledgement writing logic required by a packet handler.
+// The packet is checked for correctness including asserting that the packet was
+// sent and received on clients which are counterparties for one another.
+// If no acknowledgement exists for the given packet, then a commiment of the acknowledgement
```
Comment (AdityaSripal) [inline modules/core/packet-server/keeper/keeper.go:196]:
```suggestion
// If no acknowledgement exists for the given packet, then a commitment of the acknowledgement
```

File: modules/core/packet-server/keeper/keeper.go [hunk 4]
```diff
@@ -230,16 +241,19 @@ func (k Keeper) WriteAcknowledgement(
...
 	return nil
 }
 
-// TODO: add godoc
+// AcknowledgePacket implements the acknowledgement processing logic required by a packet handler.
+// The packet is checked for correctness including asserting that the packet was
+// sent and received on clients which are counterparties for one another.
+// If no packet commitment exists, a no-op error is returned, otherwise
+// the acknowledgement provided is verified to have been stored by the counterparty.
+// If successful the packet commitment is deleted and the packet has completed its lifecycle.
```
Comment (AdityaSripal) [inline modules/core/packet-server/keeper/keeper.go:256]:
```suggestion
// If successful, the packet commitment is deleted and the packet has completed its lifecycle.
```

File: modules/core/packet-server/keeper/keeper.go [hunk 5]
```diff
@@ -298,16 +312,20 @@ func (k Keeper) AcknowledgePacket(
...
 	return packet.AppVersion, nil
 }
 
-// TODO: add godoc
+// TimeoutPacket implements the timeout logic required by a packet handler.
+// The packet is checked for correctness including asserting that the packet was
+// sent and received on clients which are counterparties for one another.
+// If no packet commitment exists, a no-op error is returned, otherwise
+// an absence proof of the packet receipt is performed to ensure that the packet
+// was never delivered to the counterparty. If successful the packet commitment
```
Comment (AdityaSripal) [inline modules/core/packet-server/keeper/keeper.go:327]:
```suggestion
// was never delivered to the counterparty. If successful, the packet commitment
```

**Review by AdityaSripal — APPROVED** [review]
Thanks @colin-axner !

**Review by crodriguezvega — APPROVED** [review]

**Pull request merged**